crates/wasi-nn/src/backend/kserve.rs

```diff
@@ -0,0 +1,1012 @@
+            )),
+            TensorType::Fp32 => KServeTensorElement::Number(
+                Number::from_f64(cursor.read_f32::<LittleEndian>().unwrap() as f64).unwrap(),
+            ),
+            TensorType::Fp64 => KServeTensorElement::Number(
+                Number::from_f64(cursor.read_f64::<LittleEndian>().unwrap()).unwrap(),
+            ),
+            TensorType::U8 => {
+                if parse_bytes_as_string {
+                    let mut s: String = String::new();
+                    let strlen = cursor
+                        .read_to_string(&mut s)
+                        .expect("Unable to read string from tensor");
+                    assert!(tensor.dimensions[1] == 1 || (tensor.dimensions[1] as usize) == strlen);
+                    KServeTensorElement::String(s)
+                } else {
+                    KServeTensorElement::Number(Number::from(cursor.read_u8().unwrap()))
+                }
+            }
+        });
+    }
+
+    if parse_bytes_as_string {
+        //It can either be 1 or the length of data, if it was specified in the tensor.
+        assert!(expected_size == 1 || expected_size == data.len())
+    } else {
+        assert_eq!(expected_size, data.len());
+    }
+    data
+}
+
+fn map_tensor_type_to_size(tensor_type: &TensorType) -> usize {
+    match tensor_type {
+        TensorType::U8 => 1,
+        TensorType::Bf16 => 2,
+        TensorType::Fp16 => 2,
+        TensorType::I32 => 4,
+        TensorType::Fp32 => 4,
+        TensorType::Fp64 => 8,
+        TensorType::I64 => 8,
+    }
+}
+
+fn map_datatype_to_tensor_type(datatype: &KServeDatatype) -> TensorType {
+    match datatype {
+        KServeDatatype::UINT8 => TensorType::U8,
+        KServeDatatype::INT32 => TensorType::I32,
+        KServeDatatype::FP16 => TensorType::Fp16,
+        KServeDatatype::FP32 => TensorType::Fp32,
+        KServeDatatype::BF16 => TensorType::Bf16,
+        _ => panic!("Unsupported operation."),
+    }
+}
+
+fn map_tensor_type_to_datatype(tensor_type: TensorType) -> KServeDatatype {
+    match tensor_type {
+        TensorType::U8 => KServeDatatype::BYTES,
+        TensorType::Bf16 => KServeDatatype::BF16,
+        TensorType::Fp16 => KServeDatatype::FP16,
+        TensorType::I32 => KServeDatatype::INT32,
+        TensorType::Fp32 => KServeDatatype::FP32,
+        TensorType::I64 => KServeDatatype::INT64,
+        TensorType::Fp64 => KServeDatatype::FP64,
+    }
+}
+
+pub struct KServeClient {
+    server_url: String,
+    #[allow(dead_code)]
+    url: Uri,
+    #[allow(dead_code)]
+    task: JoinHandle<()>,
+    authority: Authority,
+    sender: SendRequest<Full<Bytes>>,
+}
+
+impl KServeClient {
+    /// Creates a new KServe client.
+    /// TODO: Add support for HTTPS connections.
+    pub async fn new(server_url: &String) -> Self {
+        // Parse the server url
+        let url = server_url
+            .parse::<hyper::Uri>()
+            .expect("Unable to parse url.");
```

Comment:
this code panics whenever the server url is malformed or the server doesn't respond correctly, please change it to return errors instead and handle those errors appropriately where this is invoked.